### PR TITLE
[pfm][fix] patch arbitrary account passthrough

### PR DESCRIPTION
**Pull request by agouin**
https://twitter.com/SCVSecurity/status/1682329758020022272

To ensure that users cannot pass funds through arbitrary accounts on intermediate chains, use hash of pfm+sender+channel.

e2e test: https://github.com/strangelove-ventures/interchaintest/pull/678

### CODE REVIEW

**Comment by javiersuweijie** [comment]
Hey @agouin, a PR has been made that performed the same fix: https://github.com/cosmos/ibc-apps/pull/72. Would greatly appreciate to use that instead since the tests have been fixed. 

The PR also handles some edge cases when dealing with fee processing during retries.

**Comment by agouin** [comment]
Hey @javiersuweijie thanks for your PR. I did not see the prior #69 that used the module account before starting this, my apologies. I put this up as a draft so that I could test it in the [interchaintest packet forward test ](https://github.com/strangelove-ventures/interchaintest/blob/main/examples/ibc/packet_forward_test.go). I left some comments on your PR, but I think the code changes should look more like what's here in this PR.

File: middleware/packet-forward-middleware/router/ibc_middleware.go [hunk 1]
```diff
@@ -135,6 +136,17 @@ func getBoolFromAny(value any) bool {
 	return boolVal
 }
 
+// getReceiver returns the receiver address for a given channel and original sender.
+// it overrides the receiver address to be a hash of the channel/origSender so that
+// the receiver address is deterministic and can be used to identify the sender on the
+// initial chain.
+func getReceiver(channel string, originalSender string) (string, error) {
+	senderStr := fmt.Sprintf("%s/%s", channel, originalSender)
```
Comment (nicolaslara) [inline middleware/packet-forward-middleware/router/ibc_middleware.go:144]: Should we also use the port here? (in case this is used to wrap a port other than `transfer` in the future)

Comment (jtieri) [reply middleware/packet-forward-middleware/router/ibc_middleware.go:144]: ^ this makes sense to me

**Review by jtieri — COMMENTED** [review]
I agree with @nicolaslara that adding the port to the hash for the overridden receiver could future proof this.

Otherwise this lgtm. The only other change I would suggest is making note of the new behavior somewhere in the [readme](https://github.com/cosmos/ibc-apps/blob/main/middleware/packet-forward-middleware/README.md) since this is somewhat of a substantial change to how the PFM works.

File: middleware/packet-forward-middleware/router/ibc_middleware.go (hunk 1 above)
Comment (agouin) [reply middleware/packet-forward-middleware/router/ibc_middleware.go:144]: This is not necessary, since PFM is ICS-20 middleware. Also, channels are on top of a port in IBC, so in the case this was modified in the future to support non-transfer packets, the channel ID would still give us enough to cover the channel+port. https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel

**Comment by agouin** [comment]
README has been updated to explain the intermediate receiver behavior and recommended usage of the packet receiver field, since we can use this fix as a feature to detect PFM on intermediate chains.

**Review by jtieri — APPROVED** [review]
